src/Makefile.test.include

```diff
@@ -161,6 +161,7 @@ BITCOIN_TESTS += \
 
 FUZZ_SUITE_LD_COMMON +=\
  $(LIBBITCOIN_WALLET) \
+ $(LIBBITCOIN_COMMON) \
```

Comment:
All I can do here is split it into a separate commit and explain that I have no idea why it works. The [error messages](https://github.com/bitcoin/bitcoin/pull/21935#issuecomment-858214155) that lead to this were utterly useless. I have no idea what code is broken, so no idea what needs repairing. I also never worked on fuzzer code, which doesn't help.

Reply:
> All I can do here is split it into a separate commit and explain that I have no idea why it works. The [error messages](https://github.com/bitcoin/bitcoin/pull/21935#issuecomment-858214155) that lead to this were utterly useless. I have no idea what code is broken, so no idea what needs repairing. I also never worked on fuzzer code, which doesn't help.

This seems fine but I wonder what is confusing about the error messages. On the surface, the error messages seem to just be saying libbitcoin_wallet.a can't find a few external_signer.cpp symbols. Since the symbols are in libbitcoin_common.a, it makes sense that linking libbitcoin_common.a after libbitcoin_wallet.a would fix the error.

I'm also a little confused by fanquake's comment https://github.com/bitcoin/bitcoin/pull/21935/files#r651410470 about adding a new circular dependency. I don't doubt that this PR adds one, but I'm not sure I see what it is. Maybe the comment is just referring to updating the Makefile to reflect circular dependencies in the code, though.

Reply:
It looks like the linker error is simply due to the linking order. Everywhere else we have `LIBBITCOIN_WALLET`, it is always before `LIBBITCOIN_COMMON`. But if you go up to where `FUZZ_SUITE_LD_COMMON` is first set, you see that we will end up having `LIBBITCOIN_COMMON` set before `LIBBITCOIN_WALLET` which means that the linker will have problems linking things common things that the wallet uses. Because the order is correct for the other targets, we only see a linker error for `test/fuzz/fuzz`.

I would consider the following diff to be more correct than currently implemented:
```diff
diff --git a/src/Makefile.test.include b/src/Makefile.test.include
index 157d8ee2e1..fc2fd80166 100644
--- a/src/Makefile.test.include
+++ b/src/Makefile.test.include
@@ -35,11 +35,12 @@ BITCOIN_TEST_SUITE = \
   $(TEST_UTIL_H)
 
 FUZZ_SUITE_LD_COMMON = \
+ $(LIBTEST_UTIL) \
+ $(LIBTEST_FUZZ) \
  $(LIBBITCOIN_SERVER) \
+ $(LIBBITCOIN_WALLET) \
  $(LIBBITCOIN_COMMON) \
  $(LIBBITCOIN_UTIL) \
- $(LIBTEST_UTIL) \
- $(LIBTEST_FUZZ) \
  $(LIBBITCOIN_CONSENSUS) \
  $(LIBBITCOIN_CRYPTO) \
  $(LIBBITCOIN_CLI) \
@@ -159,12 +160,7 @@ BITCOIN_TESTS += \
   wallet/test/ismine_tests.cpp \
   wallet/test/scriptpubkeyman_tests.cpp
 
-# This adds libbitcoin again to fix linking external_signer_scriptpubkeyman.cpp
-# when configured with --disable-external-signer. It is unclear why this is
-# needed and may indicate a bug elsewhere.
 FUZZ_SUITE_LD_COMMON +=\
- $(LIBBITCOIN_WALLET) \
- $(LIBBITCOIN_COMMON) \
  $(SQLITE_LIBS) \
  $(BDB_LIBS)
 
```

In this diff, `LIBTEST_UTIL` and `LIBTEST_FUZZ` are moved to the top because they include `LIBBITCOIN_SERVER` and `LIBBITCOIN_COMMON`. `LIBBITCOIN_SERVER` always needs to be the first item in the linker order since it has the most dependencies.

I think, in general, the makefiles for making the fuzz and test binaries should be revisited so that the linking order is made consistent with the rest of the code and to avoid other linker order issues that may crop up in the future.